ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -804,6 +804,59 @@ public func timer(interval: NSTimeInterval, onScheduler scheduler: DateScheduler
 	}
 }
 
+/// Creates a repeating timer of the variable interval starting with `nextInterval(0)`,
+/// with a reasonable default leeway, sending updates on the given scheduler.
+///
+/// This timer will complete when `nextInterval` returns nil.
```

Comment:
@iv-mexx 
Sounds interesting! But things will get more complicated because this `SignalType` will emit timeIntervals with time-dependency. There might be a better solution from this approach, though ;)